rp2040-hal/src/pio.rs

```diff
@@ -841,6 +841,73 @@ impl<SM: ValidStateMachine> Tx<SM> {
         true
     }
 
+    /// Write a replicated u8 value to TX FIFO.
+    ///
+    /// Memory mapped register writes that are smaller than 32bits will trigger
+    /// "Narrow IO Register Write" behaviour in RP2040 - the value written will
+    /// be replicated to the rest of the register as described in
+    /// [RP2040 Datasheet: 2.1.4. - Narrow IO Register Writes][section_2_1_4]
+    ///
+    ///
+    /// This 8bit write will set all 4 bytes of the FIFO to `value`
+    /// Eg: if you write `0xBA` the value written to the the FIFO will be
+    /// `0xBABABABA`
+    ///
+    /// If you wish to write an 8bit number without replication,
+    /// use `write(my_u8 as u32)` instead.
+    ///
+    /// Returns `true` if the value was written to FIFO, `false` otherwise.
+    ///
+    /// [section_2_1_4]: https://datasheets.raspberrypi.com/rp2040/rp2040-datasheet.pdf#_narrow_io_register_writes
```

Comment:
To make `cargo doc` happy:
```suggestion
    /// [section_2_1_4]: <https://datasheets.raspberrypi.com/rp2040/rp2040-datasheet.pdf#_narrow_io_register_writes>
```